index.js

```diff
@@ -438,51 +451,71 @@ Router.prototype.find = function find (method, path, derivedConstraints) {
     }
 
     if (currentNode.kind === NODE_TYPES.WILDCARD) {
-      const decoded = sanitizedUrl.sliceParameter(pathIndex)
-      if (decoded === null) {
-        return this._onBadUrl(path.slice(pathIndex))
+      let shouldDecode = false
+      for (let paramEndIndex = pathIndex; paramEndIndex < pathLen; paramEndIndex++) {
+        if (path.charCodeAt(paramEndIndex) === 37) {
+          shouldDecode = true
+          break
+        }
+      }
+
+      let param = originPath.slice(pathIndex)
+      if (shouldDecode) {
+        param = this.decodeURIComponent(param)
+
+        if (param === null) {
+          return this._onBadUrl(originPath)
+        }
       }
 
-      if (decoded.length > maxParamLength) {
+      if (param.length > maxParamLength) {
         return null
       }
 
-      params.push(decoded)
+      params.push(param)
       pathIndex = pathLen
       continue
     }
 
     if (currentNode.kind === NODE_TYPES.PARAMETRIC) {
       let paramEndIndex = pathIndex
+      let shouldDecode = false
       for (; paramEndIndex < pathLen; paramEndIndex++) {
-        if (path.charCodeAt(paramEndIndex) === 47) {
+        const charCode = path.charCodeAt(paramEndIndex)
+        if (charCode === 47) {
           break
+        } else if (charCode === 37) {
+          shouldDecode = true
```

Comment:
```suggestion
          shouldDecode = true
          break
```

Reply:
This loop does two things at the same time.
1) Cheks if a parameter needs to be decoded
2) Finds the end of the param ("/" symbol)

If we exit the loop as you suggested, we won't find the end of the parameter.